### PR TITLE
Fix: Nesting Observables in Vtree nodes more than one level deep

### LINKED ISSUE #71: Nesting Observables in Vtree nodes more than one level deep
TLDR: `flatMapLatest` should be used instead of `flatMap` inside `transposeVTree` 

If an observable is nested inside a vtreenode like `vtree -> observable -> vtree -> observable -> vtree -> observable` and this nesting is more than 2 observables deep the transposition is not working correctly.

This is working:
```javascript
function main ({DOM}) {
  return {
    DOM: Observable.interval(1000).startWith(0).map((i) =>div('.hello-world', [
      'Super: ' + i,
      Observable.interval(100).startWith(0).map((j) =>div('.hello-world', [
         'Sub: ' + j, 
      ]))
    ]))
  };
}
```

Expected behavior is for the sub counter to increment once per 100ms and the super counter to increment once per second. The sub counter should reset to 1 each time the super counter increments.
This is working.

How if we increase the nesting level:

```javascript
function main ({DOM}) {
  return {
    DOM: Observable.interval(10000).startWith(0).map((i) =>div('.hello-world', [
      'Super:' + i,
      Observable.interval(1000).startWith(0).map((j) =>div('.hello-world', [
         'Sub: ' + j,
         Observable.interval(100).startWith(0).map((k) =>div('.hello-world', [
          'SubSub: ' + k,
          Observable.interval(10).startWith(0).map((h) =>div('.hello-world', 'SubSubSub ' + h))
        ]))  
      ]))
    ]))
  };
}
```
Expected would be that the SubSubSub counter increments at 100 Hz and resets each time the SubSub counter increments. The SubSub counter increments at 10 Hz and resets each time the Sub counter increments. The Sub counter increments at 1 Hz and the Super counter increments once each 10 seconds.

The actual behavior is that each time a Sub* counter changes it also resets it's parents. This causes a really weird flickering.

A real world use case for this is https://github.com/Cmdv/cycle-webpack-boilerplate/
Where the routing mechanism creates a tree like:

vtree$ -> vtreeDiv(contentArea) -> currentPage$ -> vtreeDiv(pageContainer) -> pageState$ -> vtreeDiv(pageContent)

Even deeper nestings are imaginable if a page gets composed of sub pages.

A simple fix for this is to use `flatMapLatest` instead of `flatMap` inside `transposeVTree`

### PR DESCRIPTION
**Pull request by laszlokorte**
fixes #71

I will update the other PR to target the master branch.

Maybe there could be a better name for the test case.

### CODE REVIEW

File: test/browser/render.js [hunk 1]
```diff
@@ -911,5 +911,45 @@ describe('Rendering', function () {
         }
       });
     });
+
+    it('should ignore newer values of old nested observables', function (done) {
+      function app() {
+        return {
+          DOM: Rx.Observable.just(
+            div([
+              div(
```
Comment (staltz) [inline test/browser/render.js]: Is this `div` on line 920 necessary?

Comment (laszlokorte) [inline test/browser/render.js]: I think we can omit it. I will remove it.

File: test/browser/render.js [hunk 2]
```diff
@@ -911,5 +911,43 @@ describe('Rendering', function () {
         }
       });
     });
+
+    it('should ignore newer values of old nested observables', function (done) {
```
Comment (Frikki) [inline test/browser/render.js]: What is an "old" nested observable?

Comment (laszlokorte) [inline test/browser/render.js]: all but the latest one :P
If a parent Observable A emits 3 child observables X, Y and Z. X and Y are the old ones, Z is the latest one. We only want the values emitted by Z.
But yeah, there might be a better name describing it.

Comment (staltz) [inline test/browser/render.js]: Old nested Observable makes total sense to me.

Comment (Frikki) [inline test/browser/render.js]: I think a test name like: "should only be concerned with values emitted from a last child observable"

Comment (laszlokorte) [inline test/browser/render.js]: Yes that might be better.

Comment (Frikki) [inline test/browser/render.js]: Or it "should ignore values emitted from previous child observables".

**Comment by staltz** [comment]
Merged. Thanks!

**Pull request closed without merging**